docs/lti-integration.md

```diff
@@ -0,0 +1,159 @@
+configuration step.
+
+## References
+
+Canvas provides installation and configuration instructions for LTI Tools. Use
+these instructions, and refer to the steps below for Code.org specific details.
+
+- [Create a new LTI Key][install-canvas] which configures Code.org as an LTI Tool
+- [Configure the LTI app][configure-canvas] to generate the Deployment ID
+
+[install-canvas]: https://community.canvaslms.com/t5/Admin-Guide/How-do-I-configure-an-LTI-key-for-an-account/ta-p/140
+[configure-canvas]: https://community.canvaslms.com/t5/Admin-Guide/How-do-I-configure-an-external-app-for-an-account-using-a-client/ta-p/202
+
+## Installing Code.org in Canvas
+
+You will follow [these instructions][install-canvas] for configuring an LTI Key,
+i.e. installing an LTI Tool in Canavs. The outcome of this installation will be
+a Client ID to provide Code.org (in [Step 4](#step-4))
+
+### Step 1
+
+1. From the Admin menu, select `Developer Keys`
+1. Click `Add Developer Key` button, and select the `LTI Key` option
+1. Enter LTI Key Settings:
+    - Key Name: `<your-key-name>`
+    - Owner Email: `<admin-owner-email>`
+    - Redirect URIs: `https://studio.code.org`
+    - Notes: `Optional: Any notes about the LTI key, such as the reason it was created.`
+
+### Step 2
+
+There are two options for entering configuration settings. The easiest is using
+JSON. Optionally, you can chose to manually enter these settings.
+
+#### Enter JSON Details
+
+The easiest way to configure the LTI Tool in Canvas is via the `Paste JSON`
+configuration. Below is the JSON you can paste directly into the
+`LTI 1.3 Configuration` field:
+
+1. Select the `Paste JSON` option from the Configuration dropdown menu
+1. Past the below JSON into the field
+1. Click `Save`
+
+```json
+{
+  "title": "Code.org",
+  "description": "Code.org LTI Tool",
+  "oidc_initiation_url": "https://studio.code.org/lti/v1/login",
+  "target_link_uri": "https://studio.code.org/home",
+  "scopes": [
+    "https://purl.imsglobal.org/spec/lti-ags/scope/lineitem",
+    "https://purl.imsglobal.org/spec/lti-ags/scope/lineitem.readonly",
+    "https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly",
+    "https://purl.imsglobal.org/spec/lti-ags/scope/score",
+    "https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly",
+    "https://canvas.instructure.com/lti/public_jwk/scope/update",
+    "https://canvas.instructure.com/lti/account_lookup/scope/show",
+    "https://canvas.instructure.com/lti-ags/progress/scope/show"
+  ],
+  "extensions": [
+    {
+      "tool_id": "code-dot-org",
+      "platform": "canvas.instructure.com",
+      "privacy_level": "public",
+      "settings": {
+        "text": "Launch Code.org",
+        "placements": [
+          {
+            "placement": "account_navigation",
+            "message_type": "LtiResourceLinkRequest",
+             "target_link_uri": "https://studio.code.org/home"            
```

Comment:
Each of these overrides the target_link_uri that may be optionally defined at the root level. This allows us to guide the user to certain URLs depending on the context.

Reply:
We also need to determine which placements we want the tool to be available in, and have them all present in this JSON, as well as defined in the manual entry section.